### PR TITLE
Upgrade go-logging to 2.0.0-rc1.

### LINKED ISSUE #1670: Upgrade github.com/op/go-logging package
## Description

Upgrade the `github.com/op/go-logging` package to the latest version.
##### The easy part

From Vagrant

```
cd $GOPATH/src/github.com/hyperledger/fabric
go get github.com/op/go-logging
govendor update github.com/op/go-logging
```
##### The hard part

The API has changed such that `Info(...)`, `Warning(...)`, `Error(...)`, etc. functions now have `Infof(...)`, `Warningf(...)`, `Errorf(...)` equivalents. These functions are referenced in many locations.

The motivations behind this upgrade are
1. To get the latest and greatest
2. Eliminate many `go vet` warnings caused by the current API

Adding a help wanted tag to this if anyone is interested. Otherwise I may tackle this.

### PR DESCRIPTION
**Pull request by genggjh**
## Description
1. Upgrade go-logging
2. Move to use the new log functions: Debugf(...), Infof(...), Warningf(...), Errorf(...), Criticalf(...), Noticef(...)
## Motivation and Context
1. To get the latest and greatest
2. Eliminate many go vet warnings caused by the current API

Fixes #1670
## How Has This Been Tested?

Run _make all_ in my local vagrant
## Checklist:
- [x] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff)
- [x] Either no new documentation is required by this change, OR I added new documentation
- [x] Either no new tests are required by this change, OR I added new tests
- [x] I have run [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports), [go vet](https://golang.org/cmd/vet/), and [golint](https://github.com/golang/lint). I have cleaned up all valid errors and warnings in code I have added or modified. These tools may generate false positives. Don't be worried about ignoring some errors or warnings. The goal is clean, consistent, and readable code.

Signed-off-by: David Geng david@esse.io

### CODE REVIEW

**Comment by christo4ferris** [comment]
@genggjh thanks for submitting this PR! This is definitely an important piece of TD that needed to be addressed.

**Comment by ghaskins** [comment]
@genggjh Thank you.  This LGTM but it has merge conflicts.  If you would kindly rebase this I think it can go in. @binhn @srderson agree?

File: core/chaincode/shim/chaincode.go [hunk 1]
```diff
@@ -958,44 +958,37 @@ func (c *ChaincodeLogger) IsEnabledFor(level LoggingLevel) bool {
...
-// Note: We're only creating the 'f' forms of the logging functions here for
-// consistency with Go language conventions around formatted I/O routines, and
-// to avoid confusion with the conventions used in the core code. It is
-// possible that some day the core code will also change from using
-// logger.Debug() to logger.Debugf() etc., and we want to protect chaincode
-// writers from that hiccup if it occurs.
-
 // Debugf logs will only appear if the ChaincodeLogger LoggingLevel is set to
 // LogDebug.
 func (c *ChaincodeLogger) Debugf(format string, args ...interface{}) {
```
Comment (christo4ferris) [inline core/chaincode/shim/chaincode.go:915]: Having this wrapper seems unnecessarily redundant now. Not necessarily suggesting that we change it in this PR, but I will create an issue for subsequent fix.

**Comment by christo4ferris** [comment]
restarted Travis otherwise, LGTM

**Comment by genggjh** [comment]
@christo4ferris Planning to create another PR to address @bcbrock's comments which is mentioned in the issue.

**Comment by genggjh** [comment]
Anyone know what wrong with this failed?
I run the the test in my local, all tests have passed.

I can find an error in the travis-ci log file:
--- FAIL: TestRangeQuery (0.00s)
    exectransaction_test.go:1353: Error starting peer listener listen tcp 0.0.0.0:40303: bind: address already in use
--- FAIL: TestGetEvent (0.00s)
    exectransaction_test.go:1423: Error starting peer listener listen tcp 0.0.0.0:40303: bind: address already in use
FAIL
coverage: 55.6% of statements

**Comment by genggjh** [comment]
reverify jenkins

**Comment by genggjh** [comment]
reverify jenkins

**Comment by genggjh** [comment]
How to re-trigger the travis ci job? looks like the 'reverify jenkins' does not work.

**Comment by genggjh** [comment]
reverify Travis

**Comment by JonathanLevi** [comment]
> How to re-trigger the travis ci job? looks like the 'reverify jenkins' does not work.

David, I learned too that this feature was disabled, btw

**Comment by genggjh** [comment]
@JonathanLevi, never mind, will rebase it.

File: core/chaincode/shim/chaincode.go (hunk 1 above)
Comment (srderson) [inline core/chaincode/shim/chaincode.go:915]: I opened https://github.com/hyperledger/fabric/issues/1815

**Comment by srderson** [comment]
@genggjh thanks so much for upgrading the logging package! LGTM. I ran tests locally and confirmed everything is passing. I also opened issue #1815 where we can discuss whether or not we want to remove the logging wrapper from chaincode.go.

File: core/ledger/ledgermgmt/snapshot_service.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	LedgerGetter LedgerGetter
+}
+
+// LedgerGetter gets the PeerLedger associated with a channel.
+type LedgerGetter interface {
+	GetLedger(cid string) ledger.PeerLedger
+}
+
+// Generate generates a snapshot request.
+func (s *SnapshotService) Generate(ctx context.Context, signedRequest *pb.SignedSnapshotRequest) (*empty.Empty, error) {
```
[the remark this answers is not in the document]
Comment (cendhu) [reply core/ledger/ledgermgmt/snapshot_service.go]: Yes. @yacovm Before coding this PR, we discussed the approach followed in the discovery service. We felt that the 
```
Generate(SignedSnapshotRequest) returns (google.protobuf.Empty) {}
Cancel(SignedSnapshotRequest) returns (google.protobuf.Empty) {}
QueryPendings(SignedSnapshotRequest) returns (QueryPendingSnapshotsResponse) {}
```
is more expressive and readable than 
```
ProcessRequest(SignedSnapshotRequest) returns (SnapshotResponse) {}
```
and `oneof` messages. 

Further, some of the existing projects (`etcd`, `tidb`), and official gRPC tutorials also have multiple RPC methods per service. At least to me, those project services were very expressive in what they provide and I didn't have to look at the messages to find that out.

File: core/ledger/snapshotgrpc/snapshot_service.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package snapshotgrpc
```
[the remark this answers is not in the document]
Comment (manish-sethi) [reply core/ledger/snapshotgrpc/snapshot_service.go:7]: >On one side, we tried to reduce the number of packages and on the other side, we are increasing it.

I beg to differ here. It was never a goal to reduce the number of packages, it was the depth of packages that we set to reduce. Number of packages or number of files with in a package does not bother much as long as the names are meaningful, it's the depth of packages (of level more than three) that causes problems

File: internal/peer/node/start.go [hunk 4]
```diff
@@ -872,6 +873,10 @@ func serve(args []string) error {
 	// Register the Endorser server
 	pb.RegisterEndorserServer(peerServer.Server(), auth)
 
+	// register the snapshot server
+	snapshotSvc := &snapshotgrpc.SnapshotService{LedgerGetter: peerInstance}
+	pb.RegisterSnapshotServer(peerServer.Server(), snapshotSvc)
```
[the remark this answers is not in the document]
Comment (manish-sethi) [reply internal/peer/node/start.go:878]: There are two places where the list of open ledgers are being maintained. One in the `LegderMgr` and another in the `peerInstance` (inside the channel objects). All over the peer code what is used is what is maintained by the `peerInstance` and the list of opened ledgers maintained in `LegderMgr` is never used (only use is that it prevents a consumer to open one ledger more than once). In fact, I was wondering why it was not removed when the channel object was introduced to enclose all channel level resources. 

Snapshot grpc service is much closer to other services as compared to ledger code, as it simply invokes APIs on ledger. Hence, I tend to inclined toward using the data from the channel object.

In addition, as far as passing down the `LedgerMgr` is concerned, that won't help. There is no exported function in `LedgerMgr` for getting access to a ledger. Only possibility is to place this code in the `ledgermgmt` package and accessing the unexported structures.

After having gone through the above, the two options in my previous [comment](https://github.com/hyperledger/fabric/pull/1683#discussion_r466001346) may make a better sense

**Pull request merged**